src/modules/evaluateExpression/evaluateExpression.ts

```diff
@@ -136,7 +141,67 @@ async function processPgSQL(queryArray: any[], queryType: string, connection: IC
   }
 }
 
-function processGraphQL(queryArray: any[], connection: object) {
-  // TO-DO -- For now just return a value that makes test work
-  return 0
+async function processGraphQL(queryArray: any[], connection: IGraphQLConnection) {
+  const query = queryArray[0]
+  const variableNames = queryArray[1]
+  const variableNodes = queryArray.slice(2, queryArray.length - 1)
+  const returnNode = queryArray[queryArray.length - 1]
+
+  const variables = zipArraysToObject(variableNames, variableNodes)
+
+  const data = await graphQLquery(query, variables, connection)
+
+  const selectedNode = extractNode(data, returnNode)
+
+  if (Array.isArray(selectedNode)) return selectedNode.map((item) => simplifyObject(item))
+  else return simplifyObject(selectedNode)
+}
+
+// Build an object from an array of field names and an array of values
+function zipArraysToObject(variableNames: string[], variableValues: any[]) {
+  const createdObject: BasicObject = {}
+  variableNames.map((name, index) => {
+    createdObject[name] = variableValues[index]
+  })
+  return createdObject
+}
+
+// Return a specific node (e.g. application.name) from a nested Object
+function extractNode(
+  data: BasicObject,
+  node: string
+): BasicObject | string | number | boolean | BasicObject[] {
+  const returnNodeArray = node.split('.')
+  return extractNodeWithArray(data, returnNodeArray)
+
+  function extractNodeWithArray(
+    data: BasicObject,
+    nodeArray: string[]
+  ): BasicObject | string | number | boolean | BasicObject[] {
+    if (nodeArray.length === 1) return data[nodeArray[0]]
+    else return extractNodeWithArray(data[nodeArray[0]], nodeArray.slice(1))
+  }
+}
+
+// If Object has only 1 field, return just the value of that field,
+// else return the whole object.
+function simplifyObject(item: number | string | boolean | BasicObject) {
+  if (typeof item === 'object' && Object.keys(item).length === 1) return Object.values(item)[0]
+  else return item
```

Comment:
```suggestion
 return (typeof item === 'object' && Object.keys(item).length === 1) ? Object.values(item)[0] : item
```

Reply:
Good call, love a nice ternary expression. ;) Looks like I can do similarly at 152 as well:
```
if (Array.isArray(selectedNode)) return selectedNode.map((item) => simplifyObject(item))
  else return simplifyObject(selectedNode)
```
to:
```
return Array.isArray(selectedNode)
    ? selectedNode.map((item) => simplifyObject(item))
    : simplifyObject(selectedNode)
```